### PR TITLE
New Upgrade Guide Structure

### PR DESCRIPTION
**Pull request by jescalan**
This PR implements a few changes to the way that the upgrade guides are structured for v5.

1. Each guide lives on its own page, sidebar has been restructured to reflect this:

  Screenshot 2024-02-12 at 3 33 30 PM

2. Guides are **generated** from [the upgrade package](https://github.com/clerk/javascript/pull/2474), pulling a mix of markdown partials and changelog items from the CLI tool.

  Screenshot 2024-02-12 at 3 35 56 PM

3. Guides are largely self-contained, meaning that shared changes that apply across multiple Clerk SDKs are, other than in the case of the retheme changes, repeated into each SDK's guide. Rationale here is that users are not interested in our SDK layering and inheritance model and primarily would like to know what changes affect their own app, which is using a single SDK in most cases. Linking around between multiple shared sections creates more confusion than clarity, even if it's not ultimately the most DRY content as a whole.

### CODE REVIEW

File: docs/manifest.json [hunk 1]
```diff
@@ -30,8 +30,14 @@
...
-      ["Implement basic Role Based Access Control with metadata", "/guides/basic-rbac"],
-      ["Hide Personal Accounts and force organizations", "/guides/force-organizations"]
+      [
+        "Implement basic Role Based Access Control with metadata",
+        "/guides/basic-rbac"
+      ],
+      [
+        "Hide Personal Accounts and force organizations",
+        "/guides/force-organizations"
+      ]
```
Comment (LekoArts) [inline docs/manifest.json:40]: Can you remove the formatting changes from this file? They'll clutter the git blame and can lead to merge conflicts with `main` if any of these things are changed.

We'll need to fix consistent formatting independent of editor setup in another PR.

File: docs/manifest.json [hunk 2]
```diff
@@ -379,13 +427,18 @@
     [
       ["Overview", "/upgrade-guides/overview"],
       ["Long term support policy", "/upgrade-guides/long-term-support"],
-      "# SDKs",
-      ["Upgrading Next.js (v4 to v5)", "/upgrade-guides/upgrading-nextjs-from-v4-to-v5"],
-      ["Upgrading from v4 to v5", "/upgrade-guides/upgrading-from-v4-to-v5"],
-      ["Upgrading from v2 to v3", "/upgrade-guides/upgrading-from-v2-to-v3"],
+      "# Upgrading v4 to v5",
+      ["Overview (to remove)", "/upgrade-guides/v4-to-v5/general"],
```
Comment (LekoArts) [inline docs/manifest.json:431]: Do you want to remove it or not? 🤔

File: docs/manifest.json [hunk 3]
```diff
@@ -379,13 +427,18 @@
...
+      ["Remix", "/upgrade-guides/v4-to-v5/remix"],
+      ["Next.js", "/upgrade-guides/v4-to-v5/nextjs"],
+      "# Upgrading v2 to v3",
+      ["Overview", "/upgrade-guides/upgrading-from-v2-to-v3"],
```
Comment (LekoArts) [inline docs/manifest.json:435]: How about this?

```suggestion
      "# Past guides",
      ["Upgrading from v2 to v3", "/upgrade-guides/upgrading-from-v2-to-v3"],
```

File: docs/upgrade-guides/v4-to-v5/nextjs.mdx [hunk 4]
```diff
@@ -0,0 +1,920 @@
...
+
+By the end of this guide, you’ll have successfully upgraded your Next.js project to use `@clerk/nextjs` v5. You’ll learn how to update your dependencies, resolve breaking changes, and find deprecations. Step-by-step instructions will lead you through the process.
+
+## Preparing to upgrade
+
+Before uprading, it's highly recommended that you update your Clerk SDKs to the latest v4 version. Some changes required for v5 can be applied incrementally to the latest v4, which should contribute to a smoother upgrading experience.
+
+After updating, look out for deprecation messages in your terminal and browser console. By resolving these deprecations you'll be able to skip many breaking changes for v5.
+
+Additionally, some of the minumum version requirements for some base dependencies have been updated such that versions that are no longer supported or are at end-of-life are no longer guaranteed to work correctly with Clerk.
```
Comment (LekoArts) [inline docs/upgrade-guides/v4-to-v5/nextjs.mdx:18]: I'd put them all into the same paragraph

File: docs/upgrade-guides/v4-to-v5/nextjs.mdx [hunk 5]
```diff
@@ -0,0 +1,920 @@
...
+- [Upgrading from 12 to 13](https://nextjs.org/docs/pages/building-your-application/upgrading/version-13)
+- [Upgrading from 13 to 14](https://nextjs.org/docs/app/building-your-application/upgrading/version-14)
+
+## Updating to v5
+
+Whenever you feel ready, go ahead and install the latest beta version of any Clerk SDKs you are using. Make sure that you are prepared to patch some breaking changes before your app will work properly, however. Some examples for how to unstall commonly used SDKs are below:
+
+<CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
+    ```bash filename="terminal"
+    npm install @clerk/nextjs@beta
```
Comment (LekoArts) [inline docs/upgrade-guides/v4-to-v5/nextjs.mdx:57]: Right now the tag is `beta-v5` so unless we change this you'd need to adjust this

File: docs/upgrade-guides/v4-to-v5/nextjs.mdx [hunk 6]
```diff
@@ -0,0 +1,920 @@
...
+    ```bash filename="terminal"
+    yarn add @clerk/nextjs@beta
+    ```
+
+    ```bash filename="terminal"
+    pnpm add @clerk/nextjs@beta
+    ```
+</CodeBlockTabs>
+
+## CLI Upgrade Helper
```
Comment (LekoArts) [inline docs/upgrade-guides/v4-to-v5/nextjs.mdx:69]:
```suggestion
## CLI upgrade helper
```

Sentence case as per styleguide

File: docs/upgrade-guides/v4-to-v5/nextjs.mdx [hunk 7]
```diff
@@ -0,0 +1,920 @@
...
+
+As of this major version Clerk provides a `@clerk/upgrade` CLI tool that you can use to ease the upgrade. A CLI prompt will ask you a couple of questions, it will scan your codebase, and afterwards you'll get a list of changes you'll need to apply to your project, as well as the locations where they need to be applied. This should catch the vast majority of the changes needed for a successful upgrade to v5, and can save a lot of time reading through changes to things that aren't actually in your project.
```
Comment (LekoArts) [inline docs/upgrade-guides/v4-to-v5/nextjs.mdx:71]: The second and third sentence are quite long and difficult to follow. Splitting them into shorter sentences should make it easier to grok. If I'm unsure about such things I normally put the stuff into https://hemingwayapp.com/ and see what it says 😅

File: docs/upgrade-guides/v4-to-v5/nextjs.mdx [hunk 8]
```diff
@@ -0,0 +1,920 @@
...
+  ```
+
+  ```bash filename="terminal"
+  pnpm dlx @clerk/upgrade
+  ```
+</CodeBlockTabs>
+
+If you are having trouble with `npx`, it's also possible to install directly with `npm i @clerk/upgrade -g`, and can then be run with the `clerk-upgrade` command.
+
+## Breaking Changes
```
Comment (LekoArts) [inline docs/upgrade-guides/v4-to-v5/nextjs.mdx:91]:
```suggestion
## Breaking changes
```

File: docs/upgrade-guides/v4-to-v5/nextjs.mdx [hunk 9]
```diff
@@ -0,0 +1,920 @@
...
+
+### Component design adjustments
+
+The new version ships with improved design and UX across all of Clerk's [UI components](/docs/components/overview). If you have used the [`appearance` prop](/docs/components/customization/overview) or tokens for a [custom theme](/docs/components/customization/overview#create-a-custom-theme), you will likely need to make some adjustments to ensure your styling is still looking great. If you're using the [localization prop](/docs/components/customization/localization) you will likely need to make adjustments to account for added or removed localization keys.
+
+[More detail on these changes &raquo;](#redesigned-components)
```
Comment (LekoArts) [inline docs/upgrade-guides/v4-to-v5/nextjs.mdx:97]: We're not really using such arrows in other places so I'd refrain from using them in a singular instance

The anchor link won't work, right? I didn't find that section in this doc

File: docs/upgrade-guides/v4-to-v5/nextjs.mdx [hunk 10]
```diff
@@ -0,0 +1,920 @@
...
+    // will now redirect to your sign in page for un-authenticated users.
+
+    return <p>This page is now protected!</p>
+}
+````
+
+</Tab>
+</Tabs>
+
+In this example, we add `auth().protect()` to a layout, and like magic, that layout and all of it's children will be gated by auth. The same logic can be applied for pages.
```
Comment (LekoArts) [inline docs/upgrade-guides/v4-to-v5/nextjs.mdx:141]:
```suggestion
In this example, you add `auth().protect()` to a layout, and like magic, that layout and all of it's children will be gated by auth. The same logic can be applied for pages.
```

File: docs/upgrade-guides/v4-to-v5/nextjs.mdx [hunk 11]
```diff
@@ -0,0 +1,920 @@
...
+
+## Changes to top-level exports
+
+As part of this release, some of the top-level exports of `@clerk/nextjs` have been changed in order to improve bundle size and tree-shaking efficiency. These changes have resulted in a ~75% reduction in build size for middleware bundles. However, you will likely need to make some changes to import paths as a result.
+
+<Callout type='info'>
+  Use the CLI tool, `npx @clerk/upgrade`, to automatically find occurences of old imports.
+</Callout>
+
+<Accordion titles={["@clerk/nextjs/server", "@clerk/nextjs/errors", "@clerk/nextjs/app-beta", "@clerk/nextjs/ssr", "@clerk/nextjs/edge-middleware", "@clerk/nextjs/edge-middlewarefiles", "@clerk/nextjs/api"]} heading="h4">
```
Comment (LekoArts) [inline docs/upgrade-guides/v4-to-v5/nextjs.mdx:359]:
```suggestion
<Accordion titles={["@clerk/nextjs/server", "@clerk/nextjs/errors", "@clerk/nextjs/app-beta", "@clerk/nextjs/ssr", "@clerk/nextjs/edge-middleware", "@clerk/nextjs/edge-middlewarefiles", "@clerk/nextjs/api"]} heading="h3">
```

File: docs/upgrade-guides/v4-to-v5/nextjs.mdx [hunk 12]
```diff
@@ -0,0 +1,920 @@
...
+As part of this change, the default URL for each of these props has been set to `/`, so if you are passing `/` explicitly to any one of the above props, that line is no longer necessary and can be removed.
+
+```diff
+- <UserButton afterSignOutUrl='/' />
++ <UserButton />
+```
+
+## Deprecation removals & housekeeping
+
+## Deprecation removals & housekeeping
```
Comment (LekoArts) [inline docs/upgrade-guides/v4-to-v5/nextjs.mdx:463]: Double Trouble Movie poster

File: docs/upgrade-guides/v4-to-v5/nextjs.mdx [hunk 13]
```diff
@@ -0,0 +1,920 @@
...
+
+As part of this major version, a large number of previously deprecated props, arugments, methods, etc have been removed. Additionally there have been some changes to things that are only used internally, or only used very rarely. It's highly unlikely that any given app will encounter any of these items, but they are all breaking changes, so they have all been documented below.
```
Comment (LekoArts) [inline docs/upgrade-guides/v4-to-v5/nextjs.mdx:465]:
```suggestion
As part of this major version, a large number of previously deprecated props, arugments, methods, etc. have been removed. Additionally there have been some changes to things that are only used internally, or only used very rarely. It's highly unlikely that any given app will encounter any of these items, but they are all breaking changes, so they have all been documented below.
```

File: docs/upgrade-guides/v4-to-v5/remix.mdx
Comment (LekoArts) [inline docs/upgrade-guides/v4-to-v5/remix.mdx:1]: Some of my comments from Next.js will also apply here

**Review by LekoArts — CHANGES_REQUESTED** [review]

File: docs/manifest.json (hunk 1 above)
Comment (jescalan) [reply docs/manifest.json:40]: I'm not sure why they are present, to be honest. I didn't enable any sort of formatter, and I am not sure how to remove them 😕

File: docs/manifest.json (hunk 2 above)
Comment (jescalan) [reply docs/manifest.json]: Keeping it!

**Review by LekoArts — APPROVED** [review]

**Pull request merged**